Provide value for self-contained to support succeeding parameters

Cake version 0.30

When using the SelfContained property on the `DotNetCorePublishSettings` object you can't add more parameters because the donet parser thrown en exception

```
var settings = new DotNetCorePublishSettings {
	Framework = targetFramework,
	Configuration = configuration,
	SelfContained = true,
	Runtime = runtime,
	OutputDirectory = outputDirDocumentService,
	MSBuildSettings = new DotNetCoreMSBuildSettings()			
};

settings.MSBuildSettings.SetVersion("1.2.0");
 
DotNetCorePublish(projectFile, settings);
```

This created following command:
`"C:/Program Files/dotnet/dotnet.exe" publish "src/xy.csproj" --output "C:/Data/xy/output" --runtime win10-x64 --framework net472 --configuration Release --self-contained /property:Version=1.2.0`

This creates the error: 
```
C:\Program Files\dotnet\sdk\2.1.500-preview-009335\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(288,31): error MSB4030: "/property:Version=1.2.0" is an invalid value for the "IsSelfContained" parameter of the "ResolvePublishAssemblies" task. The "IsSelfContained" parameter is of type "System.Boolean". 
```

Please provide the default value `true`in order to allow succeeding parameters.

--- a/src/Cake.Common.Tests/Unit/Tools/DotNetCore/Publish/DotNetCorePublisherTests.cs
+++ b/src/Cake.Common.Tests/Unit/Tools/DotNetCore/Publish/DotNetCorePublisherTests.cs
@@ -122,7 +122,7 @@ namespace Cake.Common.Tests.Unit.Tools.DotNetCore.Publish
                 var result = fixture.Run();
 
                 // Then
-                Assert.Equal("publish --output \"/Working/artifacts\" --runtime runtime1 --framework dnxcore50 --configuration Release --version-suffix rc1 --no-build --no-dependencies --no-restore --force --self-contained --source \"https://api.nuget.org/v3/index.json\" --verbosity Minimal", result.Args);
+                Assert.Equal("publish --output \"/Working/artifacts\" --runtime runtime1 --framework dnxcore50 --configuration Release --version-suffix rc1 --no-build --no-dependencies --no-restore --force --self-contained true --source \"https://api.nuget.org/v3/index.json\" --verbosity Minimal", result.Args);
             }
 
             [Fact]

--- a/src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublisher.cs
+++ b/src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublisher.cs
@@ -122,13 +122,13 @@ namespace Cake.Common.Tools.DotNetCore.Publish
             // Self contained
             if (settings.SelfContained.HasValue)
             {
+                builder.Append("--self-contained");
                 if (settings.SelfContained.Value)
                 {
-                    builder.Append("--self-contained");
+                    builder.Append("true");
                 }
                 else
                 {
-                    builder.Append("--self-contained");
                     builder.Append("false");
                 }
             }

